lib/models/ui.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const UI = require('console-ui');
+
+let singletonUI;
+
+/**
+ * The class that serves as a singleton wrapper for `console-ui`.
+ *
+ * @private
+ * @module ember-cli
+ * @class CommandLineUI
+ * @constructor
+ */
+class CommandLineUI {
+  constructor(args) {
+    let options = args || {};
+
+    if (!singletonUI) {
+      singletonUI = new UI({
+        inputStream: options.inputStream,
```

Comment:
I don't think so. We instantiate UI only once in `lib/cli/index.js` and `process.stdin` is passed via `options` from [`bin/ember`](https://github.com/ember-cli/ember-cli/blob/master/bin/ember#L44); in tests, we use a fake UI from `console-ui`.